pkg/cloud/metadata_test.go

```diff
@@ -207,6 +368,31 @@ func TestNewMetadataService(t *testing.T) {
 				if m.GetOutpostArn() != tc.outpostArn {
 					t.Fatalf("GetOutpostArn() failed: expected %v, got %v", tc.outpostArn, m.GetOutpostArn())
 				}
+			} else if !tc.isAvailable {
+				if tc.name == "success: metadata not available, used k8s api" {
+					if err != nil {
+						t.Fatalf("NewMetadataService() failed: expected no error, got %v", err)
+					}
+					if m.GetInstanceID() != "i-abcdefgh123456789" {
```

Comment:
these could be consts to facilitate hypothetical future refactoring. doesn't mucvh matter though, just nitpicking.